code/operating_system/src/README.md

```diff
@@ -0,0 +1,6 @@
+# Operating Systems
+Operating Systems are software programs that communicate with the hardware and other application programs. They are responsible for a number of functions such as resource allocation in multi-user systems, management of files and processes (a 'process' is used to refer to a program in execution), handling Input-Ouput requests by different processes and main memory management among other roles. 
+It is one of the most crucial system components of a computer and provides the basic functionality for the device, right from the boot procedure. It also has many algorithms which enable it to make decisions about CPU scheduling etc.. 
```

Comment:
Replace "it" with "An operating system". Replace "also" with "additionally". Add another example other than CPU scheduling.